hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/process/ConsumerProcess.java

```diff
@@ -0,0 +1,153 @@
+package pl.allegro.tech.hermes.consumers.supervisor.process;
+
+import org.jctools.queues.SpscArrayQueue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.SubscriptionName;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.consumers.consumer.Consumer;
+
+import java.time.Clock;
+import java.util.Objects;
+
+public class ConsumerProcess implements Runnable {
+
+    private static final Logger logger = LoggerFactory.getLogger(ConsumerProcess.class);
+
+    private final SpscArrayQueue<Signal> signals = new SpscArrayQueue<>(100);
+
+    private final Clock clock;
+
+    private final SubscriptionName subscriptionName;
+
+    private final Consumer consumer;
+
+    private final Retransmitter retransmitter;
+
+    private volatile boolean running = true;
+
+    private volatile long healtcheckRefreshTime;
+
+    public ConsumerProcess(
+            SubscriptionName subscriptionName,
+            Consumer consumer,
+            Retransmitter retransmitter,
+            Clock clock
+    ) {
+        this.subscriptionName = subscriptionName;
+        this.consumer = consumer;
+        this.retransmitter = retransmitter;
+        this.clock = clock;
+        this.healtcheckRefreshTime = clock.millis();
+    }
+
+    @Override
+    public void run() {
+        try {
+            Thread.currentThread().setName("consumer-" + subscriptionName);
+
+            start();
+            while (running) {
+                consumer.consume(() -> processSignals());
+            }
+            stop();
+
+        } finally {
+            Thread.currentThread().setName("consumer-released-thread");
+        }
+    }
+
+    public ConsumerProcess accept(Signal signal) {
+        this.signals.add(signal);
+        return this;
+    }
+
+    public long healtcheckRefreshTime() {
+        return healtcheckRefreshTime;
+    }
+
+    private void processSignals() {
```

Comment:
In my opinion we should handle exceptions from processing signals here, especially those that could be thrown from updating topic or subscription.